example/fly_multiple_drones/fly_multiple_drones.cpp

```diff
@@ -0,0 +1,285 @@
+
+    // Setting up the callback to monitor lat and longitude
+    telemetry->position_async([&](Telemetry::Position position) {
+        myFile << getTimeStr() << "," << (system.get_uuid()) % 100000 << ","
+               << position.relative_altitude_m << "," << position.latitude_deg << ","
+               << position.longitude_deg << "," << position.absolute_altitude_m << ", \n";
+    });
+
+    // Check if vehicle is ready to arm
+    while (telemetry->health_all_ok() != true) {
+        std::cout << "Vehicle is getting ready to arm" << std::endl;
+        sleep_for(seconds(1));
+    }
+
+    // Import Mission items from QGC plan
+    Mission::mission_items_t mission_items;
+    Mission::Result import_res = Mission::import_qgroundcontrol_mission(mission_items, qgc_plan);
+    handle_mission_err_exit(import_res, "Failed to import mission items: ");
+
+    if (mission_items.size() == 0) {
+        std::cerr << "No missions! Exiting..." << std::endl;
+        exit(EXIT_FAILURE);
+    }
+    std::cout << "Found " << mission_items.size() << " mission items in the given QGC plan."
+              << std::endl;
+
+    {
+        std::cout << "Uploading mission..." << std::endl;
+        // Wrap the asynchronous upload_mission function using std::future.
+        auto prom = std::make_shared<std::promise<Mission::Result>>();
+        auto future_result = prom->get_future();
+        mission->upload_mission_async(mission_items,
+                                      [prom](Mission::Result result) { prom->set_value(result); });
+
+        const Mission::Result result = future_result.get();
+        handle_mission_err_exit(result, "Mission upload failed: ");
+        std::cout << "Mission uploaded." << std::endl;
+    }
+
+    std::cout << "Arming..." << std::endl;
+    const Action::Result arm_result = action->arm();
+    handle_action_err_exit(arm_result, "Arm failed: ");
+    std::cout << "Armed." << std::endl;
+
+    // Before starting the mission subscribe to the mission progress.
+    mission->subscribe_progress([&](int current, int total) {
+        std::cout << "Mission status update, VehicleID: " << system.get_uuid() % 100000 << " --> "
+                  << current << " / " << total << std::endl;
+    });
+
+    {
+        std::cout << "Starting mission." << std::endl;
+        auto prom = std::make_shared<std::promise<Mission::Result>>();
+        auto future_result = prom->get_future();
+        mission->start_mission_async([prom](Mission::Result result) {
+            prom->set_value(result);
+            std::cout << "Started mission." << std::endl;
+        });
+
+        const Mission::Result result = future_result.get();
+        handle_mission_err_exit(result, "Mission start failed: ");
+    }
+
+    while (!mission->mission_finished()) {
+        sleep_for(seconds(1));
+    }
+
+    // Wait for some time.
+    sleep_for(seconds(5));
+
+    // Mission complete. Landing now
+    std::cout << "Landing at last node..." << std::endl;
```

Comment:
How do you know it's landing here? I think this depends on your mission, otherwise you have to do `action->land()`.

Reply:
Changed it to return to Launch